ams/ams-server/src/main/java/com/netease/arctic/ams/server/optimize/AbstractOptimizePlan.java

```diff
@@ -113,16 +114,25 @@ protected List<BasicOptimizeTask> collectTasks(List<String> partitions) {
 
   private void accumulateFileCount(List<BasicOptimizeTask> newTasks) {
     int newFileCnt = 0;
+    long newFileSize = 0;
     for (BasicOptimizeTask optimizeTask : newTasks) {
-      int taskFileCnt = optimizeTask.getBaseFileCnt() + optimizeTask.getDeleteFileCnt() +
-          optimizeTask.getInsertFileCnt() + optimizeTask.getPosDeleteFileCnt();
-      newFileCnt += taskFileCnt;
+      newFileCnt += getTaskFileCount(optimizeTask);
+      newFileSize += getTaskFileSize(optimizeTask);
     }
     this.collectFileCnt += newFileCnt;
+    this.collectFileSize += newFileSize;
+  }
+
+  protected long getTaskFileSize(BasicOptimizeTask task) {
+    return task.getBaseFileSize() + task.getInsertFileSize() + task.getDeleteFileSize() + task.getPosDeleteFileSize();
+  }
+
+  protected int getTaskFileCount(BasicOptimizeTask task) {
+    return task.getBaseFileCnt() + task.getDeleteFileCnt() + task.getInsertFileCnt() + task.getPosDeleteFileCnt();
   }
 
   private boolean reachMaxFileCount() {
```

Comment:
I recommend changing the method name to `reachOptimizeLimit`.

Reply:
OK, I changed it.